x-pack/plugins/rollup/public/crud_app/sections/job_list/job_table/job_table.test.js

```diff
@@ -0,0 +1,193 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Pager } from '@elastic/eui';
+
+import { registerTestBed } from '../../../../../__jest__/utils';
+import { getJobs } from '../../../../../fixtures';
+import { rollupJobsStore } from '../../../store';
+import { JobTable } from './job_table';
+
+const defaultProps = {
+  jobs: [],
+  pager: new Pager(20, 10, 1),
+  filter: '',
+  sortField: '',
+  isSortAscending: false,
+  openDetailPanel: () => {},
+  closeDetailPanel: () => {},
+  filterChanged: () => {},
+  pageChanged: () => {},
+  pageSizeChanged: () => {},
+  sortChanged: () => {},
+};
+
+const initTestBed = registerTestBed(JobTable, defaultProps, rollupJobsStore);
+
+describe('<JobTable />', () => {
+  describe('table rows', () => {
+    const totalJobs = 5;
+    const jobs = getJobs(totalJobs);
+    const openDetailPanel = jest.fn();
+    const { findTestSubject } = initTestBed({ jobs, openDetailPanel });
+    const tableRows = findTestSubject('jobTableRow');
+
+    it('should create 1 table row per job', () => {
+      expect(tableRows.length).toEqual(totalJobs);
+    });
+
+    it('should create the expected 8 columns for each row', () => {
+      const expectedColumns = [
+        'id',
+        'status',
+        'indexPattern',
+        'rollupIndex',
+        'rollupDelay',
+        'dateHistogramInterval',
+        'groups',
+        'metrics'
+      ];
+
+      const tableColumns = expectedColumns
+        .map(id => {
+          const col = findTestSubject(`jobTableHeaderCell-${id}`);
+          col.__id__ = id;
+          return col;
+        })
+        .filter(col => !!col.length)
+        .map(col => col.__id__);
+
+      expect(tableColumns).toEqual(expectedColumns);
+    });
+
+    it('should set the correct job value in each row cell', () => {
+      const fields = [
+        'id',
+        'indexPattern',
+        'rollupIndex',
+        'rollupDelay',
+        'dateHistogramInterval',
+      ];
+      const row = tableRows.first();
+      const job = jobs[0];
+      const getCellText = (field) => row.find(`[data-test-subj="jobTableCell-${field}"]`).hostNodes().text();
+
+      // Simple fields
+      fields.forEach((field) => {
+        const cellText = getCellText(field);
+        expect(cellText).toEqual(job[field]);
+      });
+
+      // Status
+      const cellStatusText = getCellText('status');
+      expect(job.status).toEqual('stopped'); // make sure the job status *is* "stopped"
+      expect(cellStatusText).toEqual('Stopped');
+
+      // Groups
+      const expectedJobGroups = ['histogram', 'terms'].reduce((text, field) => {
+        if (job[field].length) {
+          return text
+            ? `${text}, ${field}`
+            : field.replace(/^\w/, char => char.toUpperCase());
+        }
+        return text;
+      }, '');
```

Comment:
It makes me raise an eyebrow when our tests replicate the logic in the code being tested _but_ I don't really have a better suggestion. I suppose we could hardcode our fixtures instead of dynamically generating them? 🤷‍♂️

Reply:
Yeah, I understand. But I guess what we are doing here is preventing an unexpected change on the component. Just like a snapshot would do. What do you mean by _hardcoding our fixtures_ ?

Reply:
Sorry, I phrased that wrong. What I meant to say was that we could just hardcode our assertion here:

```js
expect(cellGroupsText).toEqual("Histogram, terms");
```

That way we're not duplicating the implementation, and it's easier to read and understand what we expect.

Reply:
Good point, that makes sense.